crates/bevy_ecs/src/schedule/run_criteria.rs

```diff
@@ -44,6 +44,19 @@ pub enum ShouldRun {
     NoAndCheckAgain,
 }
 
+impl std::ops::Not for ShouldRun {
```

Comment:
I'm not certain that this should be the `Not` operator. I think an associated function called 'invert' or something might make more sense.

(Also I don't think rust-analyzer jumps to `Not` implementations yet (requires at least rust-analyzer/rust-analyzer#4558 iirc), so the weird factor is quite high and it's not very inspectible)

Reply:
Could it have both and then one just calls another?